mpitests/test_apply.py

```diff
@@ -0,0 +1,410 @@
+        commSize = comm.Get_size()
+        num_global_wires = commSize.bit_length() - 1
+        num_local_wires = num_wires - num_global_wires
+
+        state_vector = np.zeros(1 << num_wires).astype(np.complex128)
+        expected_output_cpu = np.zeros(1 << num_wires).astype(np.complex128)
+        local_state_vector = np.zeros(1 << num_local_wires).astype(np.complex128)
+        local_expected_output_cpu = np.zeros(1 << num_local_wires).astype(np.complex128)
+
+        state_vector = createRandomInitState(num_wires)
+
+        comm.Scatter(state_vector, local_state_vector, root=0)
+        dev_cpu = qml.device("default.qubit", wires=num_wires, c_dtype=np.complex128)
+
+        @qml.qnode(dev_cpu)
+        def circuit():
+            qml.QubitStateVector(par, wires=Wires)
+            return qml.state()
+
+        expected_output_cpu = circuit()
+        comm.Scatter(expected_output_cpu, local_expected_output_cpu, root=0)
+
+        dev_gpumpi = qml.device(
+            "lightning.gpu", wires=num_wires, mpi=True, c_dtype=np.complex128
+        )
+
+        @qml.qnode(dev_gpumpi)
+        def circuit_mpi():
+            qml.QubitStateVector(par, wires=Wires)
+            return qml.state()
+
+        local_state_vector = circuit_mpi()
+        assert np.allclose(local_state_vector, local_expected_output_cpu, atol=tol, rtol=0)
+
+    test_dev_reset = [
+        (qml.QubitStateVector, np.array([1 / np.sqrt(2), 1 / np.sqrt(2)]), [0]),
+    ]
+
+    @pytest.mark.parametrize("operation, par, Wires", test_dev_reset)
+    def test_dev_reset(self, tol, operation, par, Wires):
+        num_wires = numQubits
+        comm = MPI.COMM_WORLD
+        commSize = comm.Get_size()
+        num_global_wires = commSize.bit_length() - 1
+        num_local_wires = num_wires - num_global_wires
+
+        state_vector = np.zeros(1 << num_wires).astype(np.complex128)
+        expected_output_cpu = np.zeros(1 << num_wires).astype(np.complex128)
+        local_state_vector = np.zeros(1 << num_local_wires).astype(np.complex128)
+        local_expected_output_cpu = np.zeros(1 << num_local_wires).astype(np.complex128)
+
+        state_vector = createRandomInitState(num_wires)
+
+        comm.Scatter(state_vector, local_state_vector, root=0)
+        dev_cpu = qml.device("default.qubit", wires=num_wires, c_dtype=np.complex128)
+
+        dev_cpu.reset()
+        @qml.qnode(dev_cpu)
+        def circuit():
+            qml.PauliX(wires=[0])
+            qml.PauliX(wires=[0])
+            return qml.state()
+
+        expected_output_cpu = circuit()
+        comm.Scatter(expected_output_cpu, local_expected_output_cpu, root=0)
+
+        dev_gpumpi = qml.device(
+            "lightning.gpu", wires=num_wires, mpi=True, c_dtype=np.complex128
+        )
+        dev_gpumpi.reset()
+        @qml.qnode(dev_gpumpi)
+        def circuit_mpi():
+            qml.PauliX(wires=[0])
+            qml.PauliX(wires=[0])
+            return qml.state()
+        dev_gpumpi.reset()
```

Comment:
Is this `reset` necessary?

Reply:
aha, the test is for reset. Please check Line 371.